staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -276,10 +277,21 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	if len(chain[0].Subject.CommonName) == 0 {
 		return nil, false, nil
 	}
-	return &authenticator.Response{
-		User: &user.DefaultInfo{
-			Name:   chain[0].Subject.CommonName,
-			Groups: chain[0].Subject.Organization,
+
+	// The credential ID is the signature of the certificate.  This will be
```

Comment:
Note that cert parsing happens per request in Kube, so any change we make here is not free - we will pay this encoding cost per request.  I think that is fine, but worth mentioning.